### PR TITLE
Issue/2137

### PR DESCRIPTION
**Pull request by DavidZeni**
Fixes bug getting catapult action when hero has ballistics and no more target to shoot

### CODE REVIEW

**Comment by DjWarmonger** [comment]
I believe Catapult shouls skip it's action if there are no walls left, no matter if player has Ballistics or not.

Also we can imagien hypothetical situation when there's Catapult with no related hero.

**Comment by DavidZeni** [comment]
Ok, I'll check and refactor a bit all actions for catapult

File: server/CGameHandler.cpp [hunk 1]
```diff
@@ -5414,26 +5414,29 @@ void CGameHandler::runBattle()
...
+					makeStackDoNothing(next);
+					continue;
+				}
+
+				if(!curOwner || curOwner->getSecSkillLevel(SecondarySkill::BALLISTICS) == 0)
 				{
 					BattleAction attack;
-					attack.destinationTile = *RandomGeneratorUtil::nextItem(attackableBattleHexes, gs->getRandomGenerator());
+					attack.destinationTile = *RandomGeneratorUtil::nextItem(attackableBattleHexes,
+																			gs->getRandomGenerator());
```
Comment (DjWarmonger) [inline server/CGameHandler.cpp]: Looks like bad formatting?

File: server/CGameHandler.cpp [hunk 2]
```diff
@@ -5414,26 +5414,29 @@ void CGameHandler::runBattle()
 				continue;
 			}
 
-			if(next->getCreature()->idNumber == CreatureID::CATAPULT && (!curOwner || curOwner->getSecSkillLevel(SecondarySkill::BALLISTICS) == 0)) //catapult, hero has no ballistics
+			if( next->getCreature()->idNumber == CreatureID::CATAPULT )
```
Comment (alexvins) [inline server/CGameHandler.cpp]: No spaces after "(" and before ")", please. http://wiki.vcmi.eu/index.php?title=Coding_guidelines

**Pull request merged**